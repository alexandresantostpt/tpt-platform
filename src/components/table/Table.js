import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import TableBody from './body/TableBody'
import TableHeader from './header/TableHeader'

const StyledTable = styled.table`
    border-collapse: collapse;
    width: 100%;
`

const Table = ({ columns, data, onDelete, onEdit, onRowClick, onUnarchive }) => (
    <StyledTable>
        <TableHeader columns={columns} />
        <TableBody
            columns={columns}
            data={data}
            onDelete={onDelete}
            onEdit={onEdit}
            onRowClick={onRowClick}
            onUnarchive={onUnarchive}
        />
    </StyledTable>
)

Table.defaultProps = {
    onDelete: null,
    onEdit: null,
    onRowClick: null,
    onUnarchive: null
}

Table.propTypes = {
    columns: PropTypes.arrayOf(PropTypes.string).isRequired,
    data: PropTypes.arrayOf(PropTypes.object).isRequired,
    onDelete: PropTypes.func,
    onEdit: PropTypes.func,
    onRowClick: PropTypes.func,
    onUnarchive: PropTypes.func
}

export default Table
