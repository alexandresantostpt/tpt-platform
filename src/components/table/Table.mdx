---
menu: Table
name: Table
route: /table
---
import { Playground, PropsTable } from 'docz'

import Table from './Table'

import '@cfg/scripts'
import '@cfg/style'

# Table

<PropsTable of={ Table }/>

## Basic Usage

```
<Table columns={ ARRAY_OF_COLUMNS_NAME } data={ ARRAY_OF_OBJECT }/>
```

<Playground>
    <Table
        columns={ [ 'status', 'clientName', 'destiny', 'date' ] }
        data={
            [
                { clientName: 'Sra. Camila Santos', date: '20/04/2018', destiny: 'Itália, Florença', status: 'Em viagem' },
                { clientName: 'Sra. Camila Santos', date: '20/04/2018', destiny: 'Itália, Florença', status: 'Em viagem' },
                { clientName: 'Sra. Camila Santos', date: '20/04/2018', destiny: 'Itália, Florença', status: 'Em viagem' }
            ]
        }
    />
</Playground>
