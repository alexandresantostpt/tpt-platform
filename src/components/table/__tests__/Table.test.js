import React from 'react'
import renderer from 'react-test-renderer'

import Table from '@components/table/Table'

describe('Tests for Table component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer
            .create(
                <Table
                    columns={['columnA', 'columnB', 'columnC']}
                    data={[{ columnA: 'Column A', columnB: 'Column B', columnC: 'Column C' }]}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
