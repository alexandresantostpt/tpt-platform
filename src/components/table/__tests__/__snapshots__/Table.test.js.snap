// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Table component Should rendering without crash  1`] = `
<table
  className="Table__StyledTable-y4is3j-0 fabprq"
>
  <thead>
    <tr
      className="TableHeaderRow__StyledTableHeaderRow-bvb8hb-0 Nmgc"
    >
      <th
        className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 NeDht"
        onClick={[Function]}
      >
        labels.columnA
         
        <span
          className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 cmjVJQ"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
      </th>
      <th
        className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 NeDht"
        onClick={[Function]}
      >
        labels.columnB
         
        <span
          className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 cmjVJQ"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
      </th>
      <th
        className="TableHeaderColumn__StyledTableHeaderColumn-stotn-0 NeDht"
        onClick={[Function]}
      >
        labels.columnC
         
        <span
          className="icon icon-arrow-down Icon__StyledIcon-crr1cg-0 cmjVJQ"
          disabled={false}
          fontSize="1.75rem"
          onClick={null}
        />
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
